Add new e2e test inserting woo blocks

veljkho opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce-quality/issues/646

This test is to cover adding all Woo blocks to a new page, but except default cart and checkout and also mini-cart since that was part of another test.

While the collection block is in (beta) I would skip it from adding.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Local testing: `pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/create-woocommerce-blocks.spec.js --headed`

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

veljkho committed:
Add new e2e test inserting woo blocks

veljkho committed:
Add changelog

veljkho committed:
Include also product price

adimoldovan reviewed (changes requested):
Thanks @veljkho for adding this!

> This can be later extended with some front-end tests or multiple different flows.

Can you please create one or more tasks for any follow-up work you think is important? I fear it won't get done otherwise.

Also, can you add a simple check that the page can be published without error? I think at least that simple check should be ok for this test.

veljkho committed:
Add check after publishing the page

veljkho commented:
Hi @adimoldovan 
Thanks for the review!

> Can you please create one or more tasks for any follow-up work you think is important? I fear it won't get done otherwise.

I am not really sure why I wrote that, so I removed it. I think it may be possible to do front-end checks, but they can get complicated and not much help in the end. This one is for the merchant area, so it should be fine. I would instead add separate tests for each block because there is much to test there and not just verify all blocks rendered. We already started with some shopper side checks like [product filter by price](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/tests/shopper/shop-products-filter-by-price.spec.js).

We should revise all the tests and decide if we need another project to enhance the existing once again.

> Also, can you add a simple check that the page can be published without error? 

Sure thing, I was thinking about it. Added a new commit containing the check.

adimoldovan reviewed (approved):
Looks good. Thanks @veljkho!
